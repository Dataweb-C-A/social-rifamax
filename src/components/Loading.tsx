import { Text, Loader, createStyles, Group } from '@mantine/core'

interface ILoader {
  full?: boolean,
  message?: string,
  children?: React.ReactNode,
  withBorder?: boolean,
  color?: string,
  scheme?: "light" | "dark",
  margin?: string,
  backgroundBlur?: boolean,
}

function Loading({
  full = false,
  message,
  withBorder = false,
  backgroundBlur = false,
  color,
  scheme = "light",
  margin,
  children
}: ILoader) {
  const useStyles = createStyles((theme) => ({
    wrapper: {
      backgroundColor: scheme === 'dark' || theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0],
      backdropFilter: backgroundBlur ? 'blur(10px)' : 'none',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: full ? '100vh' : '100%',
      width: '100%',
      flexDirection: 'column',
    },
    group: {
      border: withBorder ? `1px solid ${scheme == "dark" || theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]}` : 'none',
      borderRadius: theme.radius.sm,
      padding: 20,
      margin: margin,
      backgroundColor: scheme === 'dark' || theme.colorScheme === "dark" ? theme.colors.dark[5] : theme.colors.gray[0],
    },
    message: {
      color: scheme === "dark" || theme.colorScheme === 'dark' ? theme.white : theme.black,
    }
  }))

  const { classes } = useStyles()
  return (
    <div className={classes.wrapper}>
      <Group className={classes.group}>
        <Loader size={48} color={color} />
        {message && (<Text className={classes.message} mt={-3} fw={350} fz={18}>{message}</Text>)}
      </Group>
      {children}
    </div>
  )
}

export default Loading