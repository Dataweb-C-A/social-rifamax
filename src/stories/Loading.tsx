import { lazy, ReactNode } from 'react';

const LoadingSP = lazy(() => import('../components/Loading'));

interface ILoading {
  /**
   * Do you want to be full screen?
   */
  full?: boolean;
  /**
   * Any message that needs to be displayed
   */
  message?: string;
  /**
   * Do you want to set any element inside?
   */
  children?: ReactNode;
  /**
   * Do you like borders?
   */
  withBorder?: boolean;
  /**
   * Optional background blur
   */
  backgroundBlur?: boolean;
  /**
   * Adds colors to loader
   */
  color?: string;
  /**
   * Turn on dark mode
   */
  scheme?: "light" | "dark";
  /**
   * Margin for the loader container
   */
  margin?: string;
}

/**
 * Primary UI component for user interaction
 */
export const Loading = ({
  full = false,
  message,
  withBorder = false,
  backgroundBlur = false,
  color,
  children,
  margin,
  scheme,
  ...props
}: ILoading) => {
  return (
    <LoadingSP
      full={full}
      message={message}
      color={color}
      withBorder={withBorder}
      backgroundBlur={backgroundBlur}
      scheme={scheme}
      margin={margin}
      {...props}
    >
      {children}
    </LoadingSP>
  );
};
