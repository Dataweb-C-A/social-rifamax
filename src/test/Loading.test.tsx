import renderer from 'react-test-renderer'
import Loading from '../components/Loading'

it('render loading component', () => {
  const component = renderer.create(
    <Loading
      withBorder
      message='Cargando'
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})