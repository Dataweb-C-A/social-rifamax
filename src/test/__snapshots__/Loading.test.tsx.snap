// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render loading component 1`] = `
<div
  className="mantine-4rr83w"
>
  <div
    className="mantine-Group-root mantine-h8qfnl"
  >
    <svg
      className="mantine-1avyp1d"
      role="presentation"
      stroke="#228be6"
      style={
        {
          "height": "3rem",
          "width": "3rem",
        }
      }
      viewBox="0 0 38 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          strokeWidth="5"
          transform="translate(2.5 2.5)"
        >
          <circle
            cx="16"
            cy="16"
            r="16"
            strokeOpacity=".5"
          />
          <path
            d="M32 16c0-9.94-8.06-16-16-16"
          >
            <animateTransform
              attributeName="transform"
              dur="1s"
              from="0 16 16"
              repeatCount="indefinite"
              to="360 16 16"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
    <div
      className="mantine-Text-root mantine-jp4nf4"
    >
      Cargando
    </div>
  </div>
</div>
`;
